src/DynamoPackages/Package.cs

```diff
@@ -358,8 +361,14 @@ internal IEnumerable<PackageAssembly> EnumerateAndLoadAssembliesInBinDirectory()
             // Use the pkg header to determine which assemblies to load and prevent multiple enumeration
             // In earlier packages, this field could be null, which is correctly handled by IsNodeLibrary
             var nodeLibraries = Header.node_libraries;
+
+            List<string> appDomainAssemblies = null;
+            var binFiles = new DirectoryInfo(BinaryDirectory).EnumerateFiles("*.dll");
+            if (binFiles.Any()) {
+                appDomainAssemblies = AppDomain.CurrentDomain.GetAssemblies().Select(x => x.GetName().Name).ToList();
```

Comment:
hopefully not too much of a performance hit here.
This will run for every loaded pkg that has at least one binary on disk.

Reply:
If I try to move this a level up, the assembly list might change (because we iterate all the types and dependencies might get loaded)

Reply:
have you considered how ALCs and packages containing extensions or binaries in other folders might affect this check?

Reply:
Also I am still curious about the name property of the assembly, does that contain the version number or not?

Reply:
I did not consider those. THanks !
` binaries in other folders might affect this check?` - CHanged the code to check if the faulty assembly is inside the package root folder.
`ALCs` - Changed the code to use TargetSite.Module.Assembly  - Gets the method(and assembly) where the exception was thrown - https://learn.microsoft.com/en-us/dotnet/api/system.exception.targetsite?view=net-8.0 (not sure why I did not see this before)
`curious about the name property of the assembly` - I am now using the equality operator on assemblies which should consider version and culture and ALCs